
# Row and column proportions

## Setup

In [1]:
import pandas as pd

## Data

In [5]:
LINK = "https://raw.githubusercontent.com/kirenz/datasets/master/loans.csv"

df = pd.read_csv(LINK)

## Contingency table with proportions

Normalize by dividing all values by the sum of values:

- If passed ‘all’ or True, will normalize over all values.

- If passed ‘index’ will normalize over each row.

- If passed ‘columns’ will normalize over each column.

- If margins is True, will also normalize margin values.

### Row proportions

In [3]:
# A contingency table with row proportions for the application type and homeownership variables.
pd.crosstab(df['application_type']	, df['homeownership'], normalize='index').round(4)*100

homeownership     mortgage    own   rent
application_type                        
individual           41.42  19.53  39.05
joint                58.06   6.45  35.48

- 39.05 corresponds to the proportion of individual applicants who rent.

### Column proportions

In [4]:
# A contingency table with row proportions for the application type and homeownership variables.
pd.crosstab(df['application_type']	, df['homeownership'], normalize='columns').round(4)*100

homeownership     mortgage    own   rent
application_type                        
individual           79.55  94.29  85.71
joint                20.45   5.71  14.29

- The value 85.71 indicates that 85.71% of renters applied as individuals for the loan. 